src/components/HURUmap/TreeView/index.js

```diff
@@ -8,47 +8,52 @@ import React, { useState } from "react";
 import useStyles from "./useStyles";
 
 import { ReactComponent as CheckIcon } from "@/pesayetu/assets/icons/checked.svg";
-import Link from "@/pesayetu/components/Link";
+import slugify from "@/pesayetu/utils/slugify";
 
 const TreeView = ({ items, expanded: expandedProps, ...props }) => {
   const [expanded, setExpanded] = useState(expandedProps);
   const classes = useStyles(props);
-  if (!items || !items.length) {
+  if (!items?.length) {
     return null;
   }
+  const handleClick = (title, expand) => {
+    document
+      .getElementById(slugify(title))
+      .scrollIntoView({ behavior: "smooth" });
+    if (expand) {
+      setExpanded(title);
+    }
+  };
 
   return (
     <div className={classes.root}>
       <MuiTreeView expanded={[expanded]}>
-        {items.map(({ children, label, path }) => (
+        {items.map((item) => (
           <TreeItem
-            key={path}
-            nodeId={path}
-            onClick={() => setExpanded(path)}
+            key={item.title}
+            nodeId={item.title}
+            onClick={() => handleClick(item.title, true)}
```

Comment:
() => creates a new function on every render... We could use [data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) to set the `id` and have a single `hancleClick` function that uses the `event` target to extract the dataset and scroll to that `id`.